### PR TITLE
Introduce Glean.js for collecting data

### LINKED ISSUE #117: Collect data using Glean.js
This will require Glean.js to be available and to have support for encrypted pings.

### PR DESCRIPTION
**Pull request by Dexterp37**
Fixes #117 

This PR adds Glean.js as a dependency and enables documentation and metric code generation at build time, using the 'glean' command.

**Important reviewer notes:**

- Glean.js will always submit data once initialized with `upload = true`. This only happens after user enrols in Rally.
- This PR exclusively stores the rally id as a metric in Glean. Other pings and metrics (to be at feature parity with the current implementation) will happen in a follow-up PR.

TODO:

- [x] Add tests.
- [x] Initialize Glean.
- [ ] ~Submit the onboarding ping.~ (Will happen as a separate PR)
- [ ] ~Collect `study_join_selected ` from [this spreadsheet](https://docs.google.com/spreadsheets/d/1K2QPkIVu_8BldT0wpsn0nRnfzxJxMnEo2NgTSYNaMf4/edit#gid=0&range=E16).~ (Will happen as a separate PR)
- [ ] Request a data review. (Will happen after this is reviewed, before merging)
- [x] Collect one metric.

Checklist for reviewer:

- [x] The description should reference a bug or github issue, if relevant.
- [x] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [x] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [x] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

**Comment by Dexterp37** [comment]
This is now blocked by rollup/plugins#843

**Review by brizental — APPROVED** [review]
From a Glean integration standpoint this LGTM. Very exciting to see this reaching this status. 🎉

**Comment by marniepw** [comment]
@rhelmer just a reminder about this one...thanks!

**Review by rhelmer — APPROVED** [review]
lgtm! I assume this is so simple because glean.js is doing most of the heavy lifting (well, and those bits aren't implemented that you mentioned in the top comment).

File: scripts/setupTaskcluster.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as exec from "child_process";
+import { promisify } from "util";
+
+// Define an async/await version of "exec".
+const execAsync = promisify(exec.exec);
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply scripts/setupTaskcluster.js]: No real reason, it was copy 🍝 !

**Comment by Dexterp37** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

This PR does not add any new data collection to the Core Add-on. Instead, it introduces Glean.js and stores the "rally id" in Glean.js.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

* Establish baselines or measure changes in product or platform quality or performance.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

Data collection currently happens using legacy telemetry in Firefox. We're migrating to Glean.

4) Can current instrumentation answer these questions?

Yes, but we need to migrate to Glean.js to further evolve Rally.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

This PR does not add any new data point. The Rally id/Pioneer id was already data-reviewed for Pioneer.

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.

Documentation is generated by Glean and included in this PR.

7) How long will this data be collected?  Choose one of the following:

* I want to permanently monitor this data. Ted Han (than@mozilla.com) will own the data (and the whole rally team).

8) What populations will you measure?

All the populations currently covered by the Rally policy.

9) If this data collection is default on, what is the opt-out mechanism for users?

Collection is only enabled after user enrolls in Rally after onboarding.

10) Please provide a general description of how you will analyze this data.

Data will be analysed via notebooks to understand how the core addon behaves in the wild.

11) Where do you intend to share the results of your analysis?

With key partners through their relative secure environments.

12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:

No. Data will be stored and analysed using internal tools.

**Comment by chutten** [comment]
DATA COLLECTION REVIEW RESPONSE:

    Is there or will there be documentation that describes the schema for the ultimate data set available publicly, complete and accurate?

Yes.

    Is there a control mechanism that allows the user to turn the data collection on and off?

Yes. This collection is part of Rally which can be turned on and off.

    If the request is for permanent data collection, is there someone who will monitor the data over time?

Yes, Ted Han is responsible.

    Using the category system of data types on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Mixed categories. This is the addition of a data collection system (Glean.js) not a specific data collection. 

    Is the data collection request for default-on or default-off?

Default off. Rally requires opt-in.

    Does the instrumentation include the addition of any new identifiers?

No.

    Is the data collection covered by the existing Firefox privacy notice?

Yes.

    Does there need to be a check-in in the future to determine whether to renew the data?

No. This collection is permanent.

---
Result: datareview+

**Pull request merged**